test/rmath.jl

```diff
@@ -116,26 +119,53 @@ end
 
 @testset "RMath" begin
     rmathcomp_tests("beta", [
-        ((1.0, 1.0), 0.01:0.01:0.99),
-        ((2.0, 3.0), 0.01:0.01:0.99),
-        ((10.0, 2.0), 0.01:0.01:0.99),
-        ((10, 2), 0.01:0.01:0.99),
+        ((0.1, 1.0), 0.0:0.01:1.0),
+        ((1.0, 1.0), 0.0:0.01:1.0),
+        ((2.0, 3.0), 0.0:0.01:1.0),
+        ((10.0, 2.0), 0.0:0.01:1.0),
+        ((10, 2), 0.0:0.01:1.0),
+        ((1f0, 1f0), 0f0:0.01f0:1f0),
+        ((1.0, 1.0), 0f0:0.01f0:1f0),
+        ((Float16(1), Float16(1)), Float16(0):Float16(0.01):Float16(1)),
+        ((1f0, 1f0), Float16(0):Float16(0.01):Float16(1)),
```

Comment:
I'd also test other types such as integers, rationals and/or a big type. Also, do we have tests somewhere mixing different input types to check that promotion works?

Reply:
Yes, I'll add integers and rationals as well. Promotions are already tested, e.g. here Float32 and Float16 and Float32 and Float64 above.

Reply:
I added more tests with integers and rationals. I also added tests for the values outside of the support.